junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java

```diff
@@ -141,4 +140,59 @@ public static void assertPackageNameIsValid(String packageName) {
 		}
 	}
 
+	/**
+	 * Collection of utilities for working with qualified names in Java.
+	 *
+	 * <h3>DISCLAIMER</h3>
+	 *
+	 * <p>These utilities are intended solely for usage within the JUnit framework
+	 * itself. <strong>Any usage by external parties is not supported.</strong>
+	 * Use at your own risk!
+	 *
+	 * @since 1.5
+	 */
+	@API(status = INTERNAL, since = "1.5")
+	static class JavaNameUtils {
+
+		private static final List<String> RESTRICTED_KEYWORDS = Arrays.asList("strictfp", "assert", "enum", "_",
+			"public", "protected", "private", "abstract", "static", "final", "transient", "volatile", "synchronized",
+			"native", "if", "else", "try", "catch", "finally", "do", "while", "for", "continue", "switch", "case",
+			"default", "break", "throw", "return", "this", "new", "super", "import", "instanceof", "goto", "const",
+			"null", "true", "false");
+
+		/**
+		 * Returns whether or not {@code name} is a syntactically
+		 * valid qualified name.
+		 *
+		 * @param name the string to check
+		 * @return {@code true} if this string is a
+		 * syntactically valid name, {@code false} otherwise.
+		 */
+		public static boolean isJavaName(String name) {
+			return isJavaIdentifier(name) && isNotRestrictedKeyword(name);
```

Comment:
If we switch to an immutable set like Marc proposed, wouldn't it be more efficient to switch the order here?